DOCS-6043 clarify docs around session tokens and their customization (#881)
clerk/clerk-docs: docs/backend-requests/resources/session-tokens.mdx

---
title: Session tokens
description: Learn about session tokens and how to validate them in your backend.
---

# Session tokens

When a user is authenticated in your application, Clerk generates a short-lived session token that you can use to authenticate requests to your backend. This token is a JSON Web Token (JWT) that contains information about the user and their session.

## Default session claims

> In every generated token, there are certain claims that are automatically included and cannot be overridden by templates. Clerk calls these "default claims". Many of these are standard claims registered by IANA. For generic information about them, refer to [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1).

Reviewer: ```suggestion
Every generated token has default claims that cannot be overridden by templates. Clerk's default claims include:
```
Reduced words, and moved the next line into this paragraph. In the next suggestions I'll add links to the relevant section of the RFC for each claim

> The default claims are:

Reviewer: ```suggestion
```

Can drop this with previous suggestion

- `azp`: authorized party - the `Origin` header that was included in the original Frontend API request made from the user. Most commonly, it will be the URL of the application. For example: `https://example.com`. This claim could be omitted if, for privacy-related reasons, `Origin` is empty or null.

> - `exp`: expiration time - the time after which the token will expire, as a Unix timestamp. Determined using the **Token lifetime** JWT template setting in the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=jwt-templates).

Reviewer: ```suggestion
- `exp`: expiration time - the time after which the token will expire, as a Unix timestamp. Determined using the **Token lifetime** JWT template setting in the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=jwt-templates). See [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.4) for more information.
```

> - `iat`: issued at - the time at which the token was issued as a Unix timestamp. For example: `1516239022`.

Reviewer: ```suggestion
- `iat`: issued at - the time at which the token was issued as a Unix timestamp. For example: `1516239022`. See [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.6) for more information.
```

> - `iss`: issuer - the Frontend API URL of your instance. For example: `https://clerk.your-site.com` for a production instance or `https://your-site.clerk.accounts.dev` for a development instance.

Reviewer: ```suggestion
- `iss`: issuer - the Frontend API URL of your instance. For example: `https://clerk.your-site.com` for a production instance or `https://your-site.clerk.accounts.dev` for a development instance. See [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.1) for more information.
```

> - `nbf`: not before - the time before which the token is considered invalid, as a Unix timestamp. Determined using the **Allowed Clock Skew** JWT template setting in the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=jwt-templates).

Reviewer: ```suggestion
- `nbf`: not before - the time before which the token is considered invalid, as a Unix timestamp. Determined using the **Allowed Clock Skew** JWT template setting in the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=jwt-templates). See [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.5) for more information.
```

- `sid`: session ID - the ID of the current session (e.g. `sess_123`).

> - `sub`: subject - the ID of the current user of the session (e.g. `user_123`).

Reviewer: ```suggestion
- `sub`: subject - the ID of the current user of the session (e.g. `user_123`). See [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.2) for more information.
```

- `act`: actor - will only be included if the user is impersonating another user. See [user impersonation](/docs/custom-flows/user-impersonation) for more information.

The following claims are only included if the user is part of an organization:

> - `org_id`: organization ID - the ID of the currently active organization that the user belongs to.

Reviewer: ```suggestion
- `org_id`: organization ID - the ID of the active organization that the user belongs to.
```
I don't think we need "currently" in any of these descriptions, unless there are claims that specifically give info about "previously active" organizations.
